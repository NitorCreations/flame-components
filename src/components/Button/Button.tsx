import * as React from 'react';

export enum ButtonTheme {
  primary = 'primary',
  secondary = 'secondary',
  unset = ''
}

interface IButtonPropTypes {
  onClick: () => void;
  theme?: ButtonTheme;
  disabled?: boolean;
  className?: string;
}

export default class Button extends React.PureComponent<IButtonPropTypes, {}> {
  static defaultProps = {
    theme: ButtonTheme.primary
  };

  public render() {
    const { onClick, theme, disabled, className } = this.props;
    return (
      <button className={`nitor-button ${theme} ${className ? className : ''}`} type="button" onClick={onClick} disabled={disabled}>
          <span>
              {this.props.children}
          </span>
      </button>
    );
  }
}