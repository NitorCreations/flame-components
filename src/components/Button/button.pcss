.flame-button {
  display: flex;
  outline: none;
  border: none;
  padding: var(--global-baseline);
  cursor: pointer;
  border-radius: var(--border-radius);
  will-change: background-color;
  transition: background-color var(--hover-transition);
  > span {
    font-size: calc(var(--global-baseline) * 2);
    user-select: none;
    letter-spacing: 0.1rem;
  }
  &.primary {
    background-color: var(--nitor-blue);
    > span {
      color: var(--white);
    }
    &:hover {
      background-color: color(var(--nitor-blue) tint(15%));
    }
  }
  &.secondary {
    background-color: var(--nitor-blue-light);
    > span {
      color: color(var(--nitor-blue-light) blend(var(--white) 90%));
    }
    &:hover {
      background-color: color(var(--nitor-blue-light) shade(15%));
    }
  }
  &:disabled {
    background-color: var(--light-gray);
    cursor: not-allowed;
    > span {
      color: var(--dark-gray);
    }
    &:hover {
      background-color: var(--light-gray);
    }
  }
}