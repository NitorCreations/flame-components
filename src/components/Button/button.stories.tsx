import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button, { ButtonTheme } from './Button';

storiesOf('Button', module)
  .add('Primary', () => (
      <Button onClick={action('clicked')} theme={ButtonTheme.primary}>
        Primary button
      </Button>
    )
  )
  .add('Secondary', () => (
      <Button onClick={action('clicked')} theme={ButtonTheme.secondary}>
        Secondary button
      </Button>
    )
  )
  .add('Disabled', () => (
      <Button onClick={action('clicked')} disabled>
        Disabled button
      </Button>
    )
  );