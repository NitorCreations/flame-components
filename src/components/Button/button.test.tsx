import * as React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it('Should be correctly disabled', () => {
    const component = shallow(<Button onClick={() => null} disabled />);

    expect(component.find('button').is('[disabled]')).toBeTruthy();
  });
});