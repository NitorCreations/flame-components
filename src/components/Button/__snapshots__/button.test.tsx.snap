// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
<button
  className="nitor-button primary "
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;

exports[`Button should render correctly 1`] = `
<button
  className="nitor-button primary "
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;
